neutone_sdk/parameter.py

```diff
@@ -12,28 +13,54 @@ class NeutoneParameterType(Enum):
     KNOB = "knob"
 
 
-class NeutoneParameter(NamedTuple):
+class NeutoneParameter(ABC):
     """
-    Define a Neutone Parameter that the user can use to control the model.
-
-    Currently only knob type parameters are supported.
+    Defines a Neutone Parameter abstract base class.
 
     The name and the description of the parameter will be shown as a tooltip
-    within the UI. `default_value` must be between 0 and 1 and will be used
-    as a default in the plugin when no presets are available.
+    within the UI. This parameter has no functionality.
     """
 
-    name: str
-    description: str
-    default_value: float
-    type: NeutoneParameterType = NeutoneParameterType.KNOB
-    used: bool = True
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        default_value: Union[int, float, str],
+        used: bool,
+        param_type: NeutoneParameterType,
+    ):
+        self.name = name
+        self.description = description
+        self.default_value = default_value
+        self.used = used
+        self.type = param_type
 
     def to_metadata_dict(self) -> Dict[str, str]:
         return {
             "name": self.name,
             "description": self.description,
-            "type": self.type.value,
-            "used": str(self.used),
             "default_value": str(self.default_value),
+            "used": str(self.used),
```

Comment:
Was there some reason all of these need to be strings?

Reply:
If I remember correctly, TorchScript doesn't like multi-typed dicts and keeping everything as one type also makes calling it from CPP easier.

Reply:
Ideally we would just export to JSON at the end of this method, but then it wouldnt be scriptable anymore. Perhaps we could do that in the constructor and save it to a field which is then returned by this method to get around that? Then we can ensure the metadata is parsable JSON since now that's not the case due to for example booleans being converted to python strings like "True" and "False"

Reply:
I think it should be possible to create an object like the `WaveformToWaveformMetadata` (in wavform_to_wavform.py) one to deal with this?